pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,271 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from __future__ import print_function
+
+import numpy as np
+import scipy.constants
+from pyiron.atomistics.master.parallel import AtomisticParallelMaster
+from pyiron_base import JobGenerator
+from sklearn.linear_model import LinearRegression
+from collections import defaultdict
+
+__author__ = "Joerg Neugebauer, Jan Janssen"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Jan Janssen"
+__email__ = "janssen@mpie.de"
+__status__ = "production"
+__date__ = "Sep 1, 2017"
+
+
+eV_div_A3_to_GPa = (
+    1e21 / scipy.constants.physical_constants["joule-electron volt relationship"][0]
+)
+
+def get_elastic_tensor_by_orientation(orientation, elastic_tensor):
+    """
+    Get elastic tensor in a given orientation
+
+    Args:
+        orientation (numpy.ndarray): 3x3 orthogonal orientation (no need to be orthonormal)
+        elastic_tensor (numpy.ndarray): 6x6 elastic tensor
+
+    Returns:
+        elastic_tensor in a given orientation
+    """
+    orientation = np.einsum('ij,i->ij', orientation, 1/np.linalg.norm(orientation, axis=-1))
+    if not np.isclose(np.linalg.det(orientation), 1):
+        raise ValueError('orientation must be an orthogonal 3x3 tensor')
+    C = np.zeros((3,3,3,3))
+    for i in range(3):
+        for j in range(3):
+            for k in range(3):
+                for l in range(3):
+                    C[i,j,k,l] = elastic_tensor[i+(i!=j)*(6-2*i-j), k+(k!=l)*(6-2*k-l)]
+    C = np.einsum('Ii,Jj,ijkl,Kk,Ll->IJKL', orientation, orientation, C, orientation, orientation)
```

Comment:
I'm fine with einsum, but a comment giving the equivalent in Matrix notation would be nice. I assume this is a rotation?

Reply:
It's a 4d tensor. As long as python is written on the 2d screen, I don't think there's a practical way to write it clearly without using indices.

Reply:
Mh, I guess on second glance it's actually clear what this one does.  My bad, then.